libcudacxx/include/cuda/std/__algorithm/unique_copy.h

```diff
@@ -122,13 +122,13 @@ template <class _InputIterator, class _OutputIterator, class _BinaryPredicate>
 _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14 _OutputIterator
 unique_copy(_InputIterator __first, _InputIterator __last, _OutputIterator __result, _BinaryPredicate __pred)
 {
-  using __algo_tag = __conditional_t<
+  using __algo_tag = conditional_t<
     _CCCL_TRAIT(is_base_of, forward_iterator_tag, __iterator_category_type<_InputIterator>),
     __unique_copy_tags::__reread_from_input_tag,
-    __conditional_t<_CCCL_TRAIT(is_base_of, forward_iterator_tag, __iterator_category_type<_OutputIterator>)
-                      && _CCCL_TRAIT(is_same, __iter_value_type<_InputIterator>, __iter_value_type<_OutputIterator>),
-                    __unique_copy_tags::__reread_from_output_tag,
-                    __unique_copy_tags::__read_from_tmp_value_tag>>;
+    conditional_t<_CCCL_TRAIT(is_base_of, forward_iterator_tag, __iterator_category_type<_OutputIterator>)
```

Comment:
this makes me curious, is there some reason to prefer `conditional_t` over `_If`?

Reply:
No really, except readability for outsiders

Reply:
I went ahead and replaced `conditional_t` with the efficient `_If` implementation

Reply:
I had to revert this, somehow a ton of code broke with the optimizations

Will open separate PRs for each of the optimizations